File: Models/UserPreference.cs
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;

namespace ShopNest.Models
{
    public class UserPreference
    {
        [Key]
        public int UserPreferenceId { get; set; }

        [Required]
        public int UserId { get; set; }

        public int? PreferredCategoryId { get; set; }

        public int? PreferredBrandId { get; set; }

        [Column(TypeName = "decimal(18,2)")]
        public decimal? MinPriceRange { get; set; }

        [Column(TypeName = "decimal(18,2)")]
        public decimal? MaxPriceRange { get; set; }

        public DateTime CreatedAt { get; set; } = DateTime.UtcNow;

        public DateTime? UpdatedAt { get; set; }

        // Navigation properties
        [ForeignKey("UserId")]
        public virtual User User { get; set; } = null!;

        [ForeignKey("PreferredCategoryId")]
        public virtual Category? PreferredCategory { get; set; }

        [ForeignKey("PreferredBrandId")]
        public virtual Brand? PreferredBrand { get; set; }
    }
}

File: app/Models/UserPreference.php
<?php

namespace App\Models;

use Illuminate\Database\Eloquent\Model;

class UserPreference extends Model
{
    protected $fillable = [
        'user_id',
        'preferred_category_id',
        'preferred_brand_id',
        'min_price_range',
        'max_price_range',
    ];

    protected $casts = [
        'min_price_range' => 'float',
        'max_price_range' => 'float',
    ];

    protected $dates = [
        'deleted_at',
    ];

    use \Illuminate\Database\Eloquent\SoftDeletes;

    public function user()
    {
        return $this->belongsTo(User::class);
    }

    public function preferredCategory()
    {
        return $this->belongsTo(Category::class, 'preferred_category_id');
    }

    public function preferredBrand()
    {
        return $this->belongsTo(Brand::class, 'preferred_brand_id');
    }
}
